Address a bunch of JET and Aqua issues

fingolfin opened the pull request:
Also add or adjust some TODO comments for things that should be moved to
other packages (to resolve type piracy warnings).

Some of the removed Singular code may actually be needed. We'll have to
inspect CI results to know for sure which (in any case, Singular will
have similar code in 0.19.0)

Waiting for new AA/Nemo/Hecke, see #3071

fingolfin committed:
Address a bunch of JET and Aqua issues

Also add or adjust some TODO comments for things that should be moved to
other packages (to resolve type piracy warnings).

thofma commented on src/NumberTheory/NmbThy.jl:
```diff
@@ -85,7 +85,7 @@ end
 norm_equation(R::Hecke.NfRelOrd{nf_elem,Hecke.NfOrdFracIdl}, a::NfAbsOrdElem{AnticNumberField,nf_elem}) = map(x -> R(evaluate(x)), norm_equation_fac_elem(R, a))
 
-function is_irreducible(a::NfAbsOrdElem{AnticNumberField,nf_elem})
+function is_irreducible(a::NfAbsOrdElem{AnticNumberField,nf_elem}) # TODO: move to Hecke
```
```suggestion
function is_irreducible(a::NfAbsOrdElem{AnticNumberField,nf_elem})
```
Not possible

fingolfin replied:
Ah, because of `solve_mixed`, I missed that. OK, then we a comment should be added to explain why need to commit this act of type piracy :-)

thofma replied:
"Type piracy" aka using julia as intended originally (extending methods).

thofma commented on src/Rings/mpoly-localizations.jl:
```diff
@@ -356,7 +356,7 @@ function prime_ideal(S::MPolyComplementOfKPointIdeal)
 end
 
-function is_prime(P::Hecke.ZZIdl) 
+function is_prime(P::Hecke.ZZIdl)  # TODO: move to Hecke
```
```suggestion
function is_prime(P::Hecke.ZZIdl)  # TODO: move to Hecke
```
Did you move this already?

fingolfin replied:
That's in https://github.com/thofma/Hecke.jl/pull/1274 which has not yet been merged, let alone removed; the marker here is meant to make it easy to grep for "TODO.*move" to find places that can now be removed...

thofma commented:
@StevellM can you look at the isometry changes?

fingolfin commented on experimental/Schemes/ProjectiveModules.jl:
```diff
@@ -16,7 +16,7 @@ function annihilator(M::SubquoModule)
 end
 
-iszero(I::Ideal) = all(x->(iszero(x)), gens(I))
+iszero(I::Ideal) = all(x->(iszero(x)), gens(I))  # TODO: move this to AA
```
Moved to AA in https://github.com/Nemocas/AbstractAlgebra.jl/pull/1494 but that has no yet been released.

fingolfin commented on src/GAP/iso_oscar_gap.jl:
```diff
@@ -495,7 +495,7 @@ end
 # Multivariate polynomial rings
 #
-@attributes AbstractAlgebra.Generic.MPolyRing # TODO: port this to AA
+@attributes AbstractAlgebra.Generic.MPolyRing # TODO: move this to AA
```
added to AA in https://github.com/Nemocas/AbstractAlgebra.jl/pull/1494 but that has not yet been released

fingolfin commented on src/Modules/UngradedModules.jl:
```diff
@@ -8182,13 +8182,6 @@ function default_ordering(F::FreeMod)
 
-##############################
-#should be in Singular.jl
-function Singular.intersection(a::Singular.smodule, b::Singular.smodule)
```
This is now in Singular (but it seems we don't actually use it)

fingolfin commented on src/Rings/FinField.jl:
```diff
@@ -18,7 +18,7 @@ end
-# move to Hecke/Nemo/AA?
+# TODO: move to Hecke/Nemo/AA?
 ###############################################################
 #TODO: consolidate with Hecke: is_primitive_root?
```
... resp. with `primitive_element` in Hecke

StevellM commented:
The changes on the part of enumeration of isometries are good to me.

fingolfin commented on experimental/Schemes/ProjectiveModules.jl:
```diff
@@ -16,7 +16,7 @@ function annihilator(M::SubquoModule)
 
-iszero(I::Ideal) = all(x->(iszero(x)), gens(I))
+iszero(I::Ideal) = all(x->(iszero(x)), gens(I))  # TODO: move this to AA
 
```
```suggestion
```

fingolfin commented on src/GAP/iso_oscar_gap.jl:
```diff
@@ -495,7 +495,7 @@ end
 #
-@attributes AbstractAlgebra.Generic.MPolyRing # TODO: port this to AA
+@attributes AbstractAlgebra.Generic.MPolyRing # TODO: move this to AA
 
```
```suggestion
```

fingolfin commented:
Just rebased this, and deleted the old broken `is_irreducible` method for which we now have a correct alternativ in AA.

The changes which @StevellM already reviewed are unchanged.

So assuming the tests pass I hope we can just merge this...

lgoettgens reviewed (approved):
no objections
